src/main.rs

```diff
@@ -1,20 +1,36 @@
-mod data;
 use crossterm::{
     cursor::{self, MoveTo},
     style::Print,
     terminal::{self, Clear, ClearType},
     QueueableCommand,
 };
 use std::io::{stdout, Write};
+use std::iter::zip;
+use std::sync::Arc;
+use std::sync::atomic::{AtomicBool, Ordering};
 use std::thread::sleep;
 use std::time::{Duration, Instant};
+use liblzma::write::XzDecoder;
 
-const FRAMES: usize = data::DATA.len();
-const HEIGHT: usize = data::DATA[0].len();
-const WIDTH: usize = data::DATA[0][0].len();
+const FRAMES: usize = 5258;         // This is the known frame count
+const HEIGHT: usize = 48;
+const WIDTH: usize = 16;
 const FPS: u64 = 24;
 
 fn main() {
+    let data_compressed = include_bytes!("source.bin");
+    let mut data = Vec::<u8>::new();
+    XzDecoder::new(&mut data).write_all(data_compressed).unwrap();
+    debug_assert!(*data.iter().max().unwrap() == 255);
```

Comment:
What's the purpose of this statement?

Reply:
I assume you asked about the ``debug_assert!()`` – it has a fun story: I fucked up the binary encoding of the source data the first time around in ``build.rs``. Due to a misplaced „``n <<= 1;``”, I only processed 7 bits of the input, thus the least significant bit never got any value (``.max()`` only returned 254). I added this check so it never happens again. I was contemplating to remove it, but I thought leaving it with ``debug_assert!()`` doesn't hurt, because it's only included in debug builds. It might be removed, I don't mind whether it stays or goes.